openshift/dynamic/client.py

```diff
@@ -15,11 +15,21 @@
 
 from openshift.dynamic.exceptions import ResourceNotFoundError, ResourceNotUniqueError, api_exception, KubernetesValidateMissing
 
+
+
 try:
     import kubernetes_validate
     HAS_KUBERNETES_VALIDATE = True
+    try:
+        class VersionNotSupportedError(kubernetes_validate.utils.VersionNotSupportedError):
+            pass
+    except AttributeError:
```

Comment:
Then you could get rid of this layer, you'd be left with a structure like:

```python
try:
    import kubernetes_validate
    HAS_KUBERNETES_VALIDATE = True
except ImportError:
    HAS_KUBERNETES_VALIDATE = False

try:
    from kubernetes_validate.utils import VersionNotSupportedError
except ImportError:
    class VersionNotSupportedError(NotImplementedError):
        pass
```

Which lets us avoid the nested exception structure and redundant class definitions.